### ISSUE TITLE
unable to run after update to 1.14.0 path to skaffold.yaml not properly built

### ISSUE THREAD

**Issue by smiccoli** [GoogleCloudPlatform/cloud-code-vscode#469]
Issue Type: <b>Bug</b>

*Context*
- Managed Dependencies: off
- Cloud SDK Version: 311.0.0
- Skaffold Version: v1.31.0
- Minikube Version: 1.13.1

Extension version: 1.14.0
VS Code version: Code 1.60.0 (e7d7e9a9348e6a8cc8c03f877d39cb72e5dfb1ff, 2021-09-01T10:41:52.311Z)
OS version: Windows_NT x64 10.0.19042
Restricted Mode: No

**Description**:

I just updated vscode and cloud code, and after the first run I also updated skaffold to the latest because requested by cloud code.

When I run cloud code it stales at `status IN PROGRESS > initialize session > Running skaffold dev...` in the new box in the sidebar.

If I look the new console output `Kubernetes: Run/Debug: status`, I get the following output
```
[Cloud Code]: Validating deployment settings...
[Cloud Code]: Checking kubectl, skaffold dependencies...
[Cloud Code]: Checking kubectl current context...
[Cloud Code]: Starting minikube if profile exists...
[Cloud Code]: Checking additional dependencies gcloud and docker...
[Cloud Code]: Setting up image registry...
[Cloud Code]: Setting up build environment...
[Cloud Code]: Checking skaffold profile...
[Cloud Code]: Checking logged-in status for the user...
[Cloud Code]: Running: skaffold dev -v info --port-forward --rpc-http-port 1052 --filename C\:\\dev\\my-project/skaffold.yaml --wait-for-deletions-max 2m0s --wait-for-connection -p dev
```

If I look the regular console output `Kubernetes: Run/Debug: detailed`, I get the following output
```
Running: skaffold dev -v info --port-forward --rpc-http-port 1052 --filename C\:\\dev\\my-project/skaffold.yaml --wait-for-deletions-max 2m0s --wait-for-connection -p dev
starting gRPC server on port 50051
starting gRPC HTTP server on port 1052
```
and no other information. just stuck there.

If I run skaffold manually I get more info
```
skaffold dev --filename C\:\\dev\\my-project/skaffold.yaml -p dev                                                                
parsing skaffold config: error parsing skaffold configuration file: read skaffold config: open C\:\\dev\\my-project/skaffold.yaml: The filename, directory name, or volume label syntax is incorrect
```

And indeed, if I write the path to skaffold.yaml manually, (e.g. `skaffold dev --filename C:\dev\my-project\skaffold.yaml -p dev`) it works as expected. so my first guess is that something changed in the way the path to the project skaffold.yaml is generated.

Im using a vscode workspace with multiple folders, one of which contains my backend services with skaffold.yaml at its root. No need to say that with previous versions it worked smoothly, unfortunately im not sure how the path to skaffold.yaml was built before

any ideas?

**Comment by vincentjocodes** [comment]
Hi @smiccoli,

Thank you for your report. When it is stuck, could you run `curl http://localhost:1052/v2/events` in any terminal (the port might have to be changed based on `--rpc-http-port 1052`)? Does anything print to the terminal? Do you see a stream of logs? Thank you.

**Comment by smiccoli** [comment]
this is a bit embarrasing, as I cant reproduce the issue anymore. Using your suggested curl I see the logs, but I can see them as well in the output panel, I dont see the `Kubernetes: Run/Debug: status` option anymore. 

Not sure if it was my impatience or my pc low in free resources, but waiting some time this morning made it to work. I still experience a different bootstrap time if i run through cloud code or straight from skaffold, but I guess this is due to some checks cloud code performs before running skaffold (such as minikube that I dont use).

I will keep an eye on this, but for the time being it should be fine. Apologies for the inconvenience

**Comment by vincentjocodes** [comment]
Hey @smiccoli,

Glad you got that working. We did notice a race condition on our side as well so we are putting a fix out for that now. And it does seem like under low CPU resources this race condition is likely to happen. We plan on doing a hotfix or release to Insider's branch. 

Thanks, and please let me us know if there are anymore issues.

**Comment by vincentjocodes** [comment]
Released a patch 1.14.1 with the fix, now it should also work on low CPU resource machines as well.

**Comment by jaekook-neonesia** [comment]
For the record, 

We had a similar issue with VSCode Remote SSH and Skaffold.
VSCode Server can't connect Skaffold gRPC Port. Others OK.
Finally, we found the reason. Our remote linux server host file didn't had the "127.0.01 localhost" config at /etc/hosts.
So, CloudCode command to VSCode Server to connect localhost:gRPCPort, then can't connect and holding(--wait-for-connect is default setting from CloudCode).
After we add l localhost config to /etc/hosts, going well.
